packages/tldraw/src/lib/shapes/geo/GeoShapeUtil.tsx

```diff
@@ -339,6 +341,39 @@ export class GeoShapeUtil extends BaseBoxShapeUtil<TLGeoShape> {
 		})
 	}
 
+	override getHandleSnapGeometry(shape: TLGeoShape): HandleSnapGeometry {
+		const geometry = this.getGeometry(shape)
+		// we only want to snap handles to the outline of the shape - not to its label etc.
+		const outline = geometry.children[0]
+		switch (shape.props.geo) {
```

Comment:
non-blocking, for a future refactor:
i'm thinking we should better type this because right now it feels like a runtime error if you forget to add a new shape here (and in `getGeometry` in this file)

something like:
```
enum GeoShapes {
	cloud = 'cloud',
	rectangle = 'rectangle',
	ellipse = 'ellipse',
	...
}

const actions: {[key in GeoShapes]: () => void} = {
	cloud: () => {
		...
	},
	triangle: () => {
		...
	},
	...
}
```

this way TS will give an error saying the object is non-exhaustive and will fall at compile time

Reply:
er, i meant `actions` for `getGeometry` - prbly something like `geometryForShape` instead in this instance

Reply:
@mimecuvalo the switch statement has exhaustive checking already. it's a common TS pattern where you add a `default` clause that tries to use a `never` type and fails if the type is not `never`. see `exhaustiveSwitchError`

Reply:
it would be cool to centralise the logic for each of these shape types a bit - they're very spread out and hard to track down right now. but yeah, this will already give a compile time error in the case you describe

Reply:
@ds300 ahhh i hadn't noticed that — great!
@SomeHats good point as well — it is hard to track down usage of things via string though